geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java

```diff
@@ -907,7 +911,7 @@ private void singletonRouteClientMessage(Conflatable conflatable,
    * collection of non-durable identifiers of clients connected to this VM
    */
   Set<ClientProxyMembershipID> getProxyIDs(Set mixedDurableAndNonDurableIDs) {
-    Set<ClientProxyMembershipID> result = new HashSet<>();
+    Set<ClientProxyMembershipID> result = new ConcurrentHashSet<>();
```

Comment:
Not seeing any requirement of making this ConcurrentHashSet; unless the returned set is modified, accessed in other parts at the same time.

Reply:
It's hard to see in the diff, but we assign the result to _clientInterestList and _clientInterestListInv which are accessed concurrently by multiple client registration threads/GII threads.